gc/base/SweepPoolManagerAddressOrderedListBase.cpp

```diff
@@ -236,10 +245,18 @@ MM_SweepPoolManagerAddressOrderedListBase::connectChunk(MM_EnvironmentBase *env,
 						previousConnectChunk->trailingFreeCandidate);
 
 				connectChunkPostProcess(chunk, sweepState, (MM_HeapLinkedFreeHeader *)previousConnectChunk->trailingFreeCandidate, previousFreeEntry);
+
+				if (memoryPool->doesNeedClearCardForConnectFreeEntry()) {
```

Comment:
This pattern repeats a few times. Not sure if it applies to all, but in short, what we need to do is override connectChunkPostProcess in SweepPoolManagerVLHGC that would first call MM_SweepPoolManagerAddressOrderedListBase and than the rest of postprocessing specific for VLHGC